metricflow/model/transformations/dbt_to_metricflow.py

```diff
@@ -50,6 +51,18 @@ def _build_dimensions(dimensions: List[str]) -> List[Dimension]:  # noqa: D
     return built_dimensions
 
 
+def _build_dimensions(dbt_metric: DbtMetric) -> List[Dimension]:  # noqa: D
+    dimensions = _dimensions_from_dbt_metric_dimensions(dimensions=dbt_metric.dimensions)
+    dimensions.append(
+        Dimension(
+            name=dbt_metric.timestamp,
+            type=DimensionType.TIME,
+            type_params=DimensionTypeParams(time_granularity=TimeGranularity.DAY),
```

Comment:
The truth is, I have no idea. Currently dbt metrics support `time_grains` for metrics which we decided we should ignore. The dbt `time_grains` are

> One or more "grains" at which the metric can be evaluated. For more information, see the "Custom Calendar" section.

And the valid grain options are `day`, `week`, `month`, `quarter`, `year`. But they aren't the definition of how the timestamp is specified. Thus it seemed the sane default was `day`

Reply:
My guess is the column types are timestamps, which means day granularity should work (since a month granularity timestamp in SQL is a timestamp pegged to midnight on the first of the month), and dbt 'grains' apply date_trunc expressions to the timestamp.

Basically, as long as dbt isn't doing something where a "month" grain is stored as a number between 1 and 12 in the source view we should be fine.

Reply:
I think it's a safe bet that the underlying columns are timestamps